File: Daily Test/Python/Stretched Matrix 15.11.2025.py
n = int(input())
mat1 = [input().split() for _ in range(n)]
x = int(input())
mat2 = [input().split() for _ in range(x)]
a = int(x / n)
for i in range(0, x, a):
    for j in range(0, x, a):
        for k in range(i, i + a):
            for l in range(j, j + a):
                if mat1[int(i / a)][int(j / a)] == mat2[k][l]:
                    continue
                else:
                    print('No')
                    exit()
print('Yes')

File: Daily Test/C++/Stretched Matrix 15.11.2025.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main() {
    int n, x;
    cin >> n;
    vector<vector<string>> mat1(n, vector<string>(n));
    for (int i = 0; i < n; ++i)
        for (int j = 0; j < n; ++j)
            cin >> mat1[i][j];
    cin >> x;
    vector<vector<string>> mat2(x, vector<string>(x));
    for (int i = 0; i < x; ++i)
        for (int j = 0; j < x; ++j)
            cin >> mat2[i][j];
    int a = x / n;
    for (int i = 0; i < x; i += a) {
        for (int j = 0; j < x; j += a) {
            for (int k = i; k < i + a; ++k) {
                for (int l = j; l < j + a; ++l) {
                    if (mat1[i / a][j / a] != mat2[k][l]) {
                        cout << "No" << endl;
                        return 0;
                    }
                }
            }
        }
    }
    cout << "Yes" << endl;
    return 0;
}